modules/openapi-generator/src/main/resources/swift5/libraries/urlsession/URLSessionImplementations.mustache

```diff
@@ -49,10 +55,25 @@ private var urlSessionStore = SynchronizedDictionary<String, URLSession>()
     {{#nonPublicApi}}internal{{/nonPublicApi}}{{^nonPublicApi}}open{{/nonPublicApi}} func createURLSession() -> URLSession {
         let configuration = URLSessionConfiguration.default
         configuration.httpAdditionalHeaders = buildHeaders()
```

Comment:
@DavidHorvath since now the `URLSession` is shared between all the requests, could you please remove this line?

Reply:
Something like this?
```    
open func createURLSession() -> URLSession {
        var hasher = Hasher()
        hasher.combine(buildHeaders())
        hasher.combine(credential)

        let hash = hasher.finalize()

        if let session = urlSessionStore[hash] {
            return session
        }

        let configuration = URLSessionConfiguration.default
        configuration.httpAdditionalHeaders = buildHeaders()

        let sessionDelegate = SessionDelegate()
        sessionDelegate.credential = credential

        let session = URLSession(configuration: configuration, delegate: sessionDelegate, delegateQueue: nil)
        urlSessionStore[hash] = session
        sessionDelegateStore[hash] = sessionDelegate

        return session
    }
```

Reply:
> @DavidHorvath since now the URLSession is shared between all the requests, could you please remove this line?

Just want to share a bit more this in other API client design.

In some clients (e.g. Java, C#), the object "ApiClient" is shared by all the Api classes (e.g. PetApi) by default. One can also create a new ApiClient instance to initialise the Api classes, e.g.
```
ApiClient apiClient = new ApiClient();
PetApi api = new PetApi(apiClient);
```

If each ApiClient instance has its own URLSession, then one can choose whether to share the same URLSession or have a separate URLSession per Api class.

Not sure if such design/approach works best in Swift so just to share more to help this work forward.

Reply:
Hi @wing328,
Here is works somehow the same way, but with the naming a bit different.

The main problem we were facing is that we are using for every request a new URLSession instance, and when you have many URLSession instances the iOS will throw an error.
`NSPOSIXErrorDomain Code=28 "No space left on device"`

So we are changing things a bit so that we have only one URLSession instance to all requests (although this is configurable and the user can change this if he wants).

So this will work more similar to the other generators, at least kotlin, that have an ApiClient instance per request, but all of them reuse the same OkHttp instance that is the equivalent of URLSession in iOS.

Reply:
Thanks for the explanation 👍